docs/maintaining.md

```diff
@@ -0,0 +1,22 @@
+# Maintaining Notes
+
+This document captures informal policies, tips, and topics useful to maintaining the Rust track.
+
+## Internal Tooling
+
+We have a number of scripts for CI tests.
+They live in `bin/` and `_test/`.
+
+## Internal Tooling Style Guide
+
+This is non-exhaustive.
+
+- Prefer snakecase for file names
+    ```sh
+    hello_world.sh
+    ```
+- Prefer BASH for scripting
+- Adopt a Unix philosophy for tooling
+  - try to accomplish tooling needs using `jq` and pipelines of minimal utilities
+- Scripts should use `#!/usr/bin/env bash` as their shebang
+  - This increases portability on NixOS and macOS because contributors may not have bash installed in `/bin/bash`.
```

Comment:
```suggestion
  - This increases portability on NixOS and macOS because contributors' preferred bash may not be installed in `/bin/bash`.
```

Precision of wording change. For example, in my macbook, I have these results:

```sh
$ /bin/bash --version
GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin20)
Copyright (C) 2007 Free Software Foundation, Inc.
$ /usr/bin/env bash --version
GNU bash, version 5.0.18(1)-release (x86_64-apple-darwin19.5.0)
Copyright (C) 2019 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
```

OSX installs an ancient bash in `/bin/bash` for what I believe are legal reasons with the licenses of newer versions. However, it's tedious and error-prone to replace it in place with a more modern version. Much simpler to install a new bash alongside it in some other location and configure the system to prefer the new version. `/usr/bin/env` knows how to respect standard system configurations, so it picks the newer version.

Reply:
Ah, thanks for explaining this history! Systems fun.